### PR TITLE
chore: move CI to custom runners

### PR DESCRIPTION
**Pull request by alecthomas**

### CODE REVIEW

**Review by deniseli — APPROVED** [review]
sweet!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ concurrency:
 jobs:
   test:
     name: Test Go
-    runs-on: ubuntu-latest
+    runs-on:
+      group: ftl
```
Comment (wesbillman) [inline .github/workflows/ci.yml:14]: Does this need to specify the `runner` name and the `group`? Or is just the `group` enough here? In other setups I've seen something like:

```yml
runs-on: my-cool-custom-runner
```

Comment (alecthomas) [reply .github/workflows/ci.yml:14]: Group is _supposedly_ [sufficient](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/running-jobs-on-larger-runners#using-groups-to-control-where-jobs-are-run).

**Comment by stuartwdouglas** [comment]
This does not seem to be working, is there some additional org / repo level config that is required? The other jobs I have seen with custom runners have all been in the internal org.

File: .github/actionlint.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# This is a bit of a hack to get actionlint to accept larger github runners
+# see https://github.com/rhysd/actionlint/issues/337#issuecomment-1722949069
+self-hosted-runner:
+  labels:
+    - ubuntu-22-04-32-core
```
Comment (wesbillman) [inline .github/actionlint.yaml:5]: @alecthomas I added this to avoid that `actionlint` error that the PR was getting.

Comment (alecthomas) [reply .github/actionlint.yaml:5]: Nice!

Comment (wesbillman) [reply .github/actionlint.yaml:5]: We might be able to remove this, and possibly save money if we get a smaller machine too: https://github.com/rhysd/actionlint/commit/64902ec49b95dbd4e88bf2f543885edb18bbff09#diff-8deffdcd8789dd0eda3f521d38ca69d2e988777fd4d50aec144761900ffcad37

Looks like `ubuntu-latest-4-cores`, `ubuntu-latest-8-cores`, and `ubuntu-latest-16-cores` might work

**Comment by alecthomas** [comment]
Holding off on merging this as some jobs are taking [upwards of 3 minutes](https://github.com/TBD54566975/ftl/actions/runs/10292853540) to even start.

**Comment by wesbillman** [comment]
I wonder if `ubuntu-latest-16-cores` would start up faster 🤔

File: .github/actionlint.yaml (hunk 2 above)
Comment (alecthomas) [reply .github/actionlint.yaml:5]: I wonder how you control the number of jobs/machine though. For this one I've explicitly set it to 8, so each job gets 4 cores minimum.

**Comment by alecthomas** [comment]
It looks like this couple of runs cost $338, so I think we'll bail on this.

image

**Pull request closed without merging**